javascript/ql/src/semmle/javascript/dataflow/Nodes.qll

```diff
@@ -462,3 +462,138 @@ DataFlow::SourceNode moduleMember(string path, string m) {
     result = DataFlow::ssaDefinitionNode(ssa)
   )
 }
+
+/**
+ * A data flow node corresponding to a class definition or a function definition
+ * acting as a class.
+ *
+ * The following patterns are recognized as classes and methods:
+ * ```
+ * class C {
+ *   method()
+ * }
+ * 
+ * function F() {}
+ *
+ * F.prototype.method = function() {}
+ * 
+ * F.prototype = {
+ *   method: function() {}
+ * }
+ * 
+ * extend(F.prototype, {
+ *   method: function() {}
+ * }); 
+ * ```
+ */
+class ClassNode extends DataFlow::SourceNode, DataFlow::ValueNode {
+  ClassNode() {
```

Comment:
Could we make this class extensible? I think it would be useful to be able to support ES5 libraries for creating classes, like https://www.npmjs.com/package/create-react-class.